Bluetooth: ISO: Fixes missing handling of broadcast ISO TX

Thalley opened the pull request:
Fixes issues when sending data with a broadcast ISO only build. This includes moving a few additional functions from being ACL-only to also work with BT_ISO builds, as well as some updates handling of number_completed_packets events and reading buffer sizes from the controller.

Thalley committed:
Bluetooth: Host: Move bt_conn_create_pdu* out of ACL group

Move the function out of the ACL group as it may be used
for broadcast ISO only builds.

er 31045c7 was merged, it was possible to use broadcast
iso without CONFIG_BT_CONN, but it did not properly handle
TX and RX as there were missing support to read the buffer
size from the controller, as well as missing support
for handling the number of completed packets event.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Fixes missing handling of broadcast ISO TX

After 31045c7 was merged, it was possible to use broadcast
iso without CONFIG_BT_CONN, but it did not properly handle
TX as there were missing support to read the buffer
size from the controller, as well as missing support
for handling the number of completed packets event.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented on subsys/bluetooth/host/conn.c:
```diff
@@ -1116,6 +1121,64 @@ uint8_t bt_conn_index(struct bt_conn *conn)
 }
 
+
+#if defined(CONFIG_NET_BUF_LOG)
+struct net_buf *bt_conn_create_pdu_timeout_debug(struct net_buf_pool *pool,
+						 size_t reserve,
+						 k_timeout_t timeout,
+						 const char *func, int line)
+#else
+struct net_buf *bt_conn_create_pdu_timeout(struct net_buf_pool *pool,
+					   size_t reserve, k_timeout_t timeout)
+#endif
```
This function was just moved

joerchan replied:
Could you make moving of functions a separate commit? Makes it easier to view the change you made, not just for the review, but also in the history later.

Thalley replied:
Sure

Thalley replied:
@joerchan Split the PR into 3 commits
1) Move of create_pdu
2) Move and update for num_complete_packets
3) Remaining TX/RX issues found

Thalley commented on subsys/bluetooth/host/conn.c:
```diff
@@ -1116,6 +1121,64 @@ uint8_t bt_conn_index(struct bt_conn *conn)
+	 */
+	__ASSERT_NO_MSG(!k_is_in_isr());
+
+	if (!pool) {
+		return NULL;
+	}
```
The earlier version would default to the `acl_tx_pool`, but since this is also used by ISO, then that doesn't really make sense (and also avoids defining the acl pool in broadcast-only scenarios).

Thalley replied:
Update: Had to return it to it's former "glory" as tests failed, but with a guard instead.

Thalley commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -469,6 +422,55 @@ static void hci_num_completed_packets(struct net_buf *buf)
 }
+#endif /* CONFIG_BT_CONN || CONFIG_BT_ISO */
+
+#if defined(CONFIG_BT_CONN)
+static void hci_acl(struct net_buf *buf)
```
I actually moved the `hci_num_completed_packets` function, but I guess git feels this was a better way to show that

Thalley committed:
Bluetooth: host: Add ISO support for num_completed_packets event

Change so that num_completed_packets event handling is also
enabled for broadcast ISO only builds. This is because sending
data on a broadcast ISO still generates this event.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Casper-Bonde-Bose reviewed (approved):
LGTM
